src/Features/Core/Portable/PullMemberUp/MembersPuller.cs

```diff
@@ -0,0 +1,408 @@
+            foreach (var analysisResult in result.MemberAnalysisResults)
+            {
+                foreach (var syntax in syntaxMap[analysisResult.Member])
+                {
+                    var originalMemberEditor = editorMap[syntax.SyntaxTree];
+
+                    if (analysisResult.Member.ContainingType.TypeKind == TypeKind.Interface)
+                    {
+                        // If we are pulling members from interface to interface, the original members should be removed.
+                        // Also don't need to worry other changes since the member is removed
+                        originalMemberEditor.RemoveNode(originalMemberEditor.Generator.GetDeclaration(syntax));
+                    }
+                    else
+                    {
+                        if (analysisResult.ChangeOriginalToNonStatic || analysisResult.ChangeOriginalToPublic)
+                        {
+                            ChangeMemberToPublicAndNonStatic(
+                                codeGenerationService, originalMemberEditor,
+                                syntax, analysisResult.Member);
+                        }
+                    }
+                }
+            }
+
+            return solutionEditor.GetChangedSolution();
+        }
+
+        private void ChangeMemberToPublicAndNonStatic(
+            ICodeGenerationService codeGenerationService,
+            DocumentEditor editor,
+            SyntaxNode memberSyntax,
+            ISymbol member)
+        {
+            var modifiers = DeclarationModifiers.From(member).WithIsStatic(false);
+            if (member is IEventSymbol eventSymbol)
+            {
+                ChangeEventToPublicAndNonStatic(
+                    codeGenerationService,
+                    editor,
+                    eventSymbol,
+                    memberSyntax,
+                    modifiers);
+            }
+            else
+            {
+                editor.SetAccessibility(memberSyntax, Accessibility.Public);
+                editor.SetModifiers(memberSyntax, modifiers);
+            }
+        }
+
+        private void ChangeEventToPublicAndNonStatic(
+            ICodeGenerationService codeGenerationService,
+            DocumentEditor editor,
+            IEventSymbol eventSymbol,
+            SyntaxNode memberSyntax,
+            DeclarationModifiers modifiers)
+        {
+            var declaration = editor.Generator.GetDeclaration(memberSyntax);
+            if (declaration.Equals(memberSyntax))
+            {
+                if ((eventSymbol.AddMethod != null && !eventSymbol.AddMethod.IsImplicitlyDeclared) ||
+                    (eventSymbol.RemoveMethod != null && !eventSymbol.RemoveMethod.IsImplicitlyDeclared))
+                {
+                    // One events with add or remove method
+                    editor.SetAccessibility(declaration, Accessibility.Public);
+                    editor.SetModifiers(declaration, modifiers);
```

Comment:
this is the same code as the else block below.  just have this be the final code in the method, and only do the special stuff for the single case we need.